spec/requests/movies_spec.rb

```diff
@@ -0,0 +1,337 @@
+        patch("/movies/#{movie.id}", params:)
+        expect(response).to redirect_to('/')
+      end
+
+      it 'returns redirect status' do
+        patch("/movies/#{movie.id}")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user with permission' do
+      before { sign_in manager }
+      it 'returns redirect status' do
+        patch("/movies/#{movie.id}", params:)
+        expect(response.status).to eq(302)
+      end
+
+      it 'redirects to movie/movie_id' do
+        patch("/movies/#{movie.id}", params:)
+        expect(response).to redirect_to action: :show, id: assigns(:movie).id
+      end
+
+      it 'updates movie' do
+        expect { patch("/movies/#{movie.id}", params:) }.to change {
+                                                              movie.reload.title
+                                                            }.from(movie.title).to(title.to_s)
+      end
+    end
+
+    context 'when invalid params' do
+      before { sign_in manager }
+      let(:duration) { 'some string' }
+
+      it "doesn't create movie" do
+        expect { patch("/movies/#{movie.id}", params:) }.not_to(change { movie.reload.title })
+      end
+
+      it 'returns unsuccessful response' do
+        patch("/movies/#{movie.id}", params:)
+        expect(response.status).to eq(422)
+      end
+
+      let(:duration) { Faker::Number.number(digits: 2).to_i }
+      let(:title) { nil }
+
+      it "doesn't create movie" do
+        expect { patch("/movies/#{movie.id}", params:) }.not_to(change { movie.reload.title })
+      end
+
+      it 'returns unsuccessful response' do
+        patch("/movies/#{movie.id}", params:)
+        expect(response.status).to eq(422)
+      end
+    end
+  end
+
+  describe 'DELETE /movies' do
+    let(:movie) { create :movie }
+    context 'when no user' do
+      it 'redirects to sign_in' do
+        delete("/movies/#{movie.id}")
+        expect(response).to redirect_to('/users/sign_in')
+      end
+
+      it 'returns redirect status' do
+        delete("/movies/#{movie.id}")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user without permission' do
+      before { sign_in user }
+      it 'redirects to root' do
+        delete("/movies/#{movie.id}")
+        expect(response).to redirect_to('/')
+      end
+
+      it 'returns redirect status' do
+        delete("/movies/#{movie.id}")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user with permission' do
+      before { sign_in manager }
+      it 'returns redirect status' do
+        delete("/movies/#{movie.id}")
+        expect(response.status).to eq(302)
+      end
+
+      it 'redirects to /movies' do
+        delete("/movies/#{movie.id}")
+        expect(response).to redirect_to('/movies')
+      end
+
+      it 'destroy movie' do
+        expect { delete("/movies/#{movie.id}") }.not_to change(Movie, :count)
+      end
```

Comment:
you say you check if the movie was destroyed but this expectation actually check if the movie was NOT destroyed, should be this:
```suggestion
      it 'destroy movie' do
        expect { delete("/movies/#{movie.id}") }.to change(Movie, :count).by(-1)
      end
```
I'm not sure about the (-1) tho